src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -327,46 +350,55 @@ bool CanIgnoreAllCollationElements(const UCollator* pColl, const UChar* lpStr, i
 
 }
 
-extern "C" ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName, SortHandle** ppSortHandle)
+ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName, SortHandle** ppSortHandle)
 {
-    assert(ppSortHandle != nullptr);
-    
-    *ppSortHandle = new (std::nothrow) SortHandle();
-    if ((*ppSortHandle) == nullptr)
+    assert(ppSortHandle != NULL);
+
+    *ppSortHandle = (SortHandle*)malloc(sizeof(SortHandle*));
```

Comment:
This should be `malloc(sizeof(SortHandle))`. Similar bug is in other places too.

Reply:
This was the main bug. Thanks, now it's all green.